packages/react-core/src/hooks/useDropZone.ts

```diff
@@ -86,17 +87,35 @@ export default function useDropZone({
     event.stopPropagation();
     setDragState('inactive');
     const files = Array.from(event.dataTransfer.files);
-    const { acceptedFiles, rejectedFiles } = filterAllowedFiles<File>(
-      files,
-      acceptedFileTypes
-    );
 
     if (isFunction(_onDrop)) {
       _onDrop(event);
     }
-    if (isFunction(onDropComplete)) {
-      onDropComplete({ acceptedFiles, rejectedFiles });
+
+    // If no DataTransfer items interface, handle as simple files
+    if (!event.dataTransfer.items) {
```

Comment:
Not as nit: Is there any reason that we would not just exclusively deal with items in the presence of them? E.g.
```ts
if (isFunction(_onDrop)) {
   _onDrop(event);
}

// replacing line 89
const { items, files } = event.dataTransfer;

// 
!items ? processDroppedItems(items) : completeDrop(files);
```

Where `completeDrop` would encapsulate the duplicated functionality of line 97-103 and be called by processDroppedItems instead of chaining then here. (I think the current processing function even already creates this flattened list of `Files`, right?)

Reply:
I'm not sure if I understand. Maybe I'm misinterpreting your first statement, but we were already only dealing with items if they existed, because conditionally called logic for handling files included an early return.

That said, I agree with your approach for simplifying the logic here. I moved the repeated ending logic into `completeDrop`. However, rather than passing `completeDrop` into `processDroppedItems`, I thought it might be clearer to just chain it as `processDroppedItems(items).then(completeDrop)`. What do you think?

Reply:
I think my confusion was actually over the shadowed `files` variable due to it being declared in this parent function but then the same variable name was used as the parameter of the `then`. I initially thought it was the same files being used in both places